sdcm/sct_events.py

```diff
@@ -50,9 +50,11 @@ def run(self):
 
             backend.setsockopt(zmq.LINGER, 0)  # pylint: disable=no-member
             frontend.setsockopt(zmq.LINGER, 0)  # pylint: disable=no-member
-
-            self.ready_event.set()
             zmq.proxy(frontend, backend)  # pylint: disable=no-member
+            if not self.wait_till_event_loop_is_working(timeout=120, number_of_events=20):
```

Comment:
is this code reachable ? i.e. as far as I remember zmq.proxy is running forever, until we stop, so what's the point of this code ?

Reply:
this would actually would run the code, and we could get rid of the event:
```diff
diff --git a/sdcm/sct_events.py b/sdcm/sct_events.py
index 02bd53a9..0a71d82a 100644
--- a/sdcm/sct_events.py
+++ b/sdcm/sct_events.py
@@ -26,7 +26,6 @@ class EventsDevice(Process):
 
     def __init__(self, log_dir):
         super(EventsDevice, self).__init__()
-        self.ready_event = Event()
         self.pub_port = Value('d', 0)
         self.sub_port = Value('d', 0)
 
@@ -51,10 +50,6 @@ class EventsDevice(Process):
             backend.setsockopt(zmq.LINGER, 0)  # pylint: disable=no-member
             frontend.setsockopt(zmq.LINGER, 0)  # pylint: disable=no-member
             zmq.proxy(frontend, backend)  # pylint: disable=no-member
-            if not self.wait_till_event_loop_is_working(timeout=120, number_of_events=20):
-                LOGGER.error("EventDevice event loop failed to deliver 20 test events with no loss")
-                raise RuntimeError("Event loop is not working")
-            self.ready_event.set()
 
         except Exception:  # pylint: disable=broad-except
             LOGGER.exception("zmq device failed")
@@ -73,10 +68,12 @@ class EventsDevice(Process):
 
         end_time = time.time() + timeout
         while time.time() < end_time:
+
             result = True
             for _ in range(number_of_events):
                 if not StartupTestEvent().publish(guaranteed=True, timeout=1, republish=3):
                     result = False
+                    print(_)
                     break
             if result:
                 return True
@@ -721,7 +718,6 @@ def start_events_device(log_dir, timeout=5):
 
     EVENTS_PROCESSES['MainDevice'] = EventsDevice(log_dir)
     EVENTS_PROCESSES['MainDevice'].start()
-    EVENTS_PROCESSES['MainDevice'].ready_event.wait(timeout=timeout)
 
     EVENTS_PROCESSES['EVENTS_FILE_LOOGER'] = EventsFileLogger(log_dir)
     EVENTS_PROCESSES['EVENTS_GRAFANA_ANNOTATOR'] = GrafanaAnnotator()
@@ -731,6 +727,10 @@ def start_events_device(log_dir, timeout=5):
     EVENTS_PROCESSES['EVENTS_GRAFANA_ANNOTATOR'].start()
     EVENTS_PROCESSES['EVENTS_GRAFANA_AGGRAGATOR'].start()
 
+    if not EVENTS_PROCESSES['MainDevice'].wait_till_event_loop_is_working(timeout=30, number_of_events=5):
+        LOGGER.error("EventDevice event loop failed to deliver 5 test events with no loss")
+        raise RuntimeError("Event loop is not working")
+
     # default filters
     EVENTS_PROCESSES['default_filter'] = []
     EVENTS_PROCESSES['default_filter'] += [DbEventsFilter(type='BACKTRACE', line='Rate-limit: supressed')]

```

Reply:
Fixed